Prioritize hgln_obs over crln_obs when extracting observer information from FITS headers

svank opened the pull request:
Fix as proposed in #7184

Thanks @ayshih for confirming this should be right!

dstansby commented on sunpy/coordinates/wcs_utils.py:
```diff
@@ -129,6 +129,7 @@ def solar_wcs_frame_mapping(wcs):
                              attrs[1] * u.deg,
                              attrs[2] * u.m,
                              **kwargs)
+            break
```
Could you add a quick comment above where the `required_attrs` dictionary is defined that the order of entries matters there?

dstansby reviewed (changes requested):
Looks 👍 , just one request for a comment, and needs a bugifx changelog entry adding.

svank commented:
@dstansby Thanks! I've made these changes.

ayshih commented on sunpy/coordinates/tests/test_wcs_utils.py:
```diff
@@ -399,3 +399,51 @@ def test_obsgeo_invalid(obsgeo):
+    """
+    When extracting the observer information from a FITS header, ensure
+    Stonyhurst (HG) coordinates are preferred over Carrington (CR) if present.
+    Going through `Map` and `WCS` seem to follow (at least slightly) different
+    code paths, so test them both.
```
```suggestion
    `Map` creates a custom FITS header with a sanitized observer location, so
    we need to also test a directly instantiated `WCS` object.
```

ayshih reviewed (changes requested):
Some text edits, otherwise looks good to me!

If I were to be super-picky, the part of the unit test that tests the prioritization in the `Map` implementation should really live under `sunpy.map.tests` and not `sunpy.coordinates.tests`, because the relevant code – how `GenericMap` prioritizes and sanitizes the observer location – all lives in `sunpy/map/mapbase.py`.

Cadair commented:
>  the part of the unit test that tests the prioritization in the Map implementation should really live under sunpy.map.tests

Given my recent foray into seeing if it's possible to run tests for subpackages without optional dependencies installed, ideally we would not have a direct reliance on map in the coordinates tests.

svank commented:
What's code review for, if not being picky? I moved the `Map` part of the test to the `map` module---there is still a pre-existing `Map` dependence in `coordinates/tests/test_wcs_utils.py`, but now I avoid further entrenching that.

Text edits are also implemented.

ayshih reviewed (approved):
Looks good to me!

ayshih commented:
Thanks, @svank!
